src/result_buffer.jl

```diff
@@ -113,7 +155,10 @@ struct TaskResultBuffer <: AbstractResultBuffer
     column_indicators::BitSetMatrix
 end
 
+# Since the chunk size if always <= 2GiB, we can never never overflow a PosLen31
+# which uses 31 bits for the position and 31 bits for the length
```

Comment:
> chunk size if always <= 2GiB

where is this enforced? I wonder if here we should assert that the <=2GiB restriction is still in place?

Reply:
ChunkedBase and NewlineLexers have loud checks for that. The main reason is exactly the PosLen31 thing.